src/librustc_codegen_utils/symbol_names.rs

```diff
@@ -219,6 +219,9 @@ fn get_symbol_hash<'a, 'tcx>(
                 .hash_stable(&mut hcx, &mut hasher);
             (&tcx.crate_disambiguator(instantiating_crate)).hash_stable(&mut hcx, &mut hasher);
         }
+
+        let is_vtable_shim = instance.is_vtable_shim();
+        is_vtable_shim.hash_stable(&mut hcx, &mut hasher);
```

Comment:
Shouldn't we be hashing something from the `Instance` *anyway*? I thought we already had overlaps in `(DefId, Substs)` between possible `Instance`s, did I imagine that?